src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -74,37 +74,145 @@ public DevBoxInstance(IDevBoxAuthService devBoxAuthService)
     }
 
     // Returns a DevBox object from a JSON object
-    public void FillFromJson(JsonElement item, string project, IDeveloperId? devId) => throw new NotImplementedException();
-
-    // ToDo: Use System.Uri
-    private (string WebURI, string RdpURI) GetRemoteLaunchURIsAsync(string boxURI)
-    {
-         return (string.Empty, string.Empty);
-    }
-
-    public ComputeSystemOperations SupportedOperations => ComputeSystemOperations.Start | ComputeSystemOperations.ShutDown;
+    public void FillFromJson(JsonElement item, string project, IDeveloperId? devId)
+    {
+        try
+        {
+            BoxURI = new Uri(item.GetProperty("uri").ToString());
+            Name = item.GetProperty("name").ToString();

```

Comment:
Are all properties must be present?
JsonElement.GetPropertty will throw an exception if property does not exist.

Reply:
Hence the try/catch. :)